### PR TITLE
Use OCaml's configure with LDFLAGS and libs from solo5 & freestanding

### PR DESCRIPTION
**Pull request by hannesm**
See individual commits for details, this exposes #101 on all platforms.

### CODE REVIEW

**Comment by hannesm** [comment]
The last force-push now patches OCaml's configure to avoid passing `-lm` for conftest. Instead, the `-lopenlibm` is used. This allows configure to execute nicely.

File: Makefile [hunk 1]
```diff
@@ -53,9 +52,6 @@ ocaml/Makefile.config: ocaml/Makefile
 	echo 'SAK_CC=cc' >> ocaml/Makefile.config
 	echo 'SAK_CFLAGS=$(OC_CFLAGS) $(OC_CPPFLAGS)' >> ocaml/Makefile.config
 	echo 'SAK_LINK=$(SAK_CC) $(SAK_CFLAGS) $(OUTPUTEXE)$(1) $(2)' >> ocaml/Makefile.config
-	echo '#define HAS_GETTIMEOFDAY' >> ocaml/runtime/caml/s.h
-	echo '#define HAS_SECURE_GETENV' >> ocaml/runtime/caml/s.h
-	echo '#define HAS_TIMES' >> ocaml/runtime/caml/s.h
```
Comment (dinosaure) [inline Makefile:58]: Why we need to remove these lines?

Comment (dinosaure) [reply Makefile:58]: ah ok I saw the commit. I will do the end to end test on Ubuntu and just a compilation on OpenBSD 7.

Comment (hannesm) [reply Makefile:58]: since OCaml's configure is now using nolibc and openlibm, it properly detects these features now. tested on 4.13.0.

**Comment by dinosaure** [comment]
It seems that this version of `ocaml-freestanding` brings some _syscalls_ on the produced `libasmrun.a` such as `getpid`. I will try to look why.

**Comment by hannesm** [comment]
On my FreeBSd system with 4.13.0 the difference I can observe is that `s.h` (generated by configure, in `ocaml/runtime/caml/s.h`) now `defines HAS_GETCWD 1`, which leads to a different conditionally compiled code in `sys.c`. I consider this behaviour as fine, `caml_sys_getcwd` will still error out. OTOH we may be able to drop the `STUB_ABORT(getcwd)` from nolibc. I can't see any `getuid` refeences.

**Comment by dinosaure** [comment]
Ok I found a final solution, we must keep:
```
        echo '#define HAS_SECURE_GETENV' >> ocaml/runtime/caml/s.h
```

If we delete it, we bring some Unix dependencies. Then, you are right about `GETTIMEOFDAY` and `HAS_TIMES`, the `configure` is able to generate them correctly.

**Comment by hannesm** [comment]
please excuse my lack of Linux testing. I force-pushed to use `-lnolibc -lopenlibm` (instead of `-lopenlibm -lnolibc`) in `Makefile` -- and now this works fine on an Ubuntu 20.04 (focal).

**Comment by dinosaure** [comment]
The `dns-resolver` can compiles :tada:, this PR is the definitive fix for #101 and we should fix #92 with that. Waiting the CI and ready to merge/cut a release.

**Comment by hannesm** [comment]
Sorry, I force-pushed again to use `sed` instead of a million of patches. This makes it more portable (though we've to be careful about new OCaml releases and may at some point need to act differently for different OCaml versions).

**Comment by dinosaure** [comment]
As we said in #101, #104 solves the initial issue on OpenBSD 7. This PR is not needed anymore so. Thanks!

**Pull request closed without merging**